src/main/java/com/emc/mongoose/metrics/context/MetricsContextImpl.java

```diff
@@ -237,4 +244,83 @@ public final String toString() {
 	public final void close() {
 		super.close();
 	}
+
+	public static ContextBuilder builder() {
+		return new ContextBuilderImpl();
+	}
+
+	private static class ContextBuilderImpl
+		implements ContextBuilder<ContextBuilder, MetricsContextImpl> {
+
+		private IntSupplier actualConcurrencyGauge;
+		private String id;
+		private OpType opType;
+		private int concurrencyLimit;
+		private int concurrencyThreshold;
+		private SizeInBytes itemDataSize;
+		private boolean stdOutColorFlag;
+		private int outputPeriodSec;
+		private String comment;
+
+		public MetricsContextImpl build() {
+			Arrays.asList(this.getClass().getDeclaredFields()).forEach(field -> {
+				try {
+					if(field.get(this) == null) {
```

Comment:
fields with a primitive type will pass this
it would be better to not use any checks than use the reflection